### PR TITLE
feat: XML host file reference variable array support

### PR DESCRIPTION
**Pull request by mm-broadcom**
Adds the grammar to support XML host and file reference variable arrays per IBM's documentation: https://www.ibm.com/docs/en/db2-for-zos/13?topic=cobol-host-variable-arrays-in#db2z_hostvariablearraycobol__title__20

## How Has This Been Tested?

Verified in VS Code with a sample program/setup and also made some unit tests as well.

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java [hunk 1]
```diff
@@ -350,4 +360,9 @@ void testBinaryHostVariableArray3() {
   void testBinaryHostVariableArray4() {
     UseCaseEngine.runTest(BINARY_ARR_TEXT4, ImmutableList.of(), ImmutableMap.of());
   }
+
+  @Test
```
Comment (slavek-kucera) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:510]: Add test cases validating
- range being 2-48 
- occurrence number being positive number <32K

File: server/engine/src/main/antlr4/org/eclipse/lsp/db2/parser/Db2SqlParser.g4 [hunk 2]
```diff
@@ -47,7 +47,7 @@ result_set_locator_variable: dbs_level_01 entry_name host_variable_usage result_
 
 tableLocators_variable: dbs_host_var_levels entry_name host_variable_usage tableLocators;
 
-lob_xml_host_variables: dbs_host_var_levels entry_name  host_variable_usage xml_as (lobWithSize | xml_lobNO_size);
+lob_xml_host_variables: dbs_host_var_levels entry_name host_variable_usage xml_as (lobWithSize | xml_lobNO_size) host_variable_array_times?;
```
Comment (slavek-kucera) [inline server/engine/src/main/antlr4/org/eclipse/lsp/db2/parser/Db2SqlParser.g4:50]: I think there should be a change to the visitors to produce that correct varibles.
```
       04  VAR                              OCCURS 12345.              
         49 VAR-LENGTH                     PIC S9(9) COMP-5.           
         49 VAR-DATA                       PIC X(10).                  
      *04 VAR USAGE IS SQL TYPE IS XML AS CLOB (10) OCCURS 12345 TIMES.
```

**Review by slavek-kucera — CHANGES_REQUESTED** [review]

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java (hunk 1 above)
Comment (mm-broadcom) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:510]: Resolved in latest commit.

File: server/engine/src/main/antlr4/org/eclipse/lsp/db2/parser/Db2SqlParser.g4 (hunk 2 above)
Comment (mm-broadcom) [reply server/engine/src/main/antlr4/org/eclipse/lsp/db2/parser/Db2SqlParser.g4:50]: Addressed in latest commit.

File: server/engine/src/main/antlr4/org/eclipse/lsp/db2/parser/Db2SqlParser.g4 [hunk 3]
```diff
@@ -47,7 +47,7 @@ result_set_locator_variable: dbs_level_01 entry_name host_variable_usage result_
 
 tableLocators_variable: dbs_host_var_levels entry_name host_variable_usage tableLocators;
 
-lob_xml_host_variables: dbs_host_var_levels entry_name  host_variable_usage xml_as (lobWithSize | xml_lobNO_size);
+lob_xml_host_variables: dbs_host_var_levels entry_name host_variable_usage xml_as (lobWithSize | xml_lobNO_size) (host_variable_array_times?);
```
Comment (slavek-kucera) [inline server/engine/src/main/antlr4/org/eclipse/lsp/db2/parser/Db2SqlParser.g4]:
```suggestion
lob_xml_host_variables: dbs_host_var_levels entry_name host_variable_usage xml_as (lobWithSize | xml_lobNO_size) host_variable_array_times?;
```

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java [hunk 4]
```diff
@@ -245,6 +245,42 @@ public class TestSqlHostVariable {
...
+                  + "       01 {$*VAR-NAME1}.\n"
+                  + "       04 {$*VAR`->VAR`->VAR-LENGTH`->VAR-DATA} USAGE IS SQL TYPE IS XML AS CLOB (10) OCCURS 12345 TIMES.\n"
+                  + "        PROCEDURE DIVISION.\n";
+
+  public static final String LOB_XML_ARR_TEXT2 =
+          "        Identification Division.\n"
+                  + "        Program-Id. 'TEST1'.\n"
+                  + "        Data Division.\n"
+                  + "         Working-Storage Section.\n"
+                  + "       01 {$*VAR`->VAR`->VAR-LENGTH`->VAR-DATA} USAGE IS SQL TYPE IS XML AS CLOB(10) OCCURS 1234 TIMES.\n"
```
Comment (slavek-kucera) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:262]:
```suggestion
                  + "       01 {$*VAR`->VAR`->VAR-LENGTH`->VAR-DATA|1} USAGE IS SQL TYPE IS XML AS CLOB(10) OCCURS 1234 TIMES.\n"
```
?

File: server/engine/src/main/antlr4/org/eclipse/lsp/db2/parser/Db2SqlParser.g4 (hunk 3 above)
Comment (mm-broadcom) [reply server/engine/src/main/antlr4/org/eclipse/lsp/db2/parser/Db2SqlParser.g4:50]: The parenthesis resolved a ZWSP issue I was getting. (ie. Zero-width space)

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java (hunk 4 above)
Comment (mm-broadcom) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:262]: This string is for a zero-error test, we should not be getting anything from diagnostics.

Comment (mm-broadcom) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:262]: Adding that will break the test. (Tried it locally first.) The point of this string is to test the locality/level not the variable.

Comment (slavek-kucera) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:262]: The level should not be 01, right?

Comment (mm-broadcom) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:262]: Correct, that is done on purpose. It is a check for when the user puts an invalid (in this instance) level on a XML file/reference variable array.

**Review by slavek-kucera — APPROVED** [review]

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java [hunk 5]
```diff
@@ -245,6 +245,42 @@ public class TestSqlHostVariable {
...
+                  + "        PROCEDURE DIVISION.\n";
+
+  public static final String LOB_XML_ARR_TEXT4 =
+          "        Identification Division.\n"
+                  + "        Program-Id. 'TEST1'.\n"
+                  + "        Data Division.\n"
+                  + "         Working-Storage Section.\n"
+                  + "       01 {$*GREET}.\n"
+                  + "          02 {$*VA`->VA`->VA-LENGTH`->VA-DATA} USAGE IS SQL TYPE IS XML AS CLOB (10) OCCURS 12345 TIMES.\n"
+                  + "        PROCEDURE DIVISION.\n"
```
Comment (slavek-kucera) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:281]:
```suggestion
                  + "        PROCEDURE DIVISION.\n"
                  + "           DISPLAY {$VA}(1).\n"
```
to verify that it is there?

Comment (mm-broadcom) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java:281]: Implemented in latest commit. Could not do it directly as `VA` needed to be `{$VA}` instead.

**Review by amanPras — APPROVED** [review]

**Pull request merged**